pulp_rpm/app/tasks/copy.py

```diff
@@ -78,51 +78,72 @@ def copy_content(config, dependency_solving):
             criteria MUST be validated before being passed to this task.
         content_pks: a list of content pks to copy from source to destination
     """
-    for entry in config:
-        source_repo_version = RepositoryVersion.objects.get(pk=entry["source_repo_version"])
-        dest_repo = RpmRepository.objects.get(pk=entry["dest_repo"])
-
-        dest_version_provided = bool(entry.get("dest_base_version"))
-        if dest_version_provided:
-            dest_repo_version = RepositoryVersion.objects.get(pk=entry["dest_base_version"])
-        else:
-            dest_repo_version = dest_repo.latest_version()
-
-        if entry.get("content"):
-            content_filter = Q(pk__in=entry.get("content"))
-        else:
-            content_filter = Q()
-
-        if not dependency_solving:
+    if not dependency_solving:
+        # No Dependency Solving Branch
+        # ============================
+        for entry in config:
+            source_repo_version = RepositoryVersion.objects.get(pk=entry["source_repo_version"])
+            dest_repo = RpmRepository.objects.get(pk=entry["dest_repo"])
+
+            dest_version_provided = bool(entry.get("dest_base_version"))
+            if dest_version_provided:
+                dest_repo_version = RepositoryVersion.objects.get(pk=entry["dest_base_version"])
+            else:
+                dest_repo_version = dest_repo.latest_version()
+
+            if entry.get("content"):
+                content_filter = Q(pk__in=entry.get("content"))
+            else:
+                content_filter = Q()
+
             content_to_copy = source_repo_version.content.filter(content_filter)
             content_to_copy |= find_children_of_content(content_to_copy, source_repo_version)
 
             base_version = dest_repo_version if dest_version_provided else None
             with dest_repo.new_version(base_version=base_version) as new_version:
                 new_version.add_content(content_to_copy)
-
-            continue
-
+    else:
         # Dependency Solving Branch
         # =========================
-        content_to_copy = {}
 
-        # TODO: add lookaside repos here
-        repo_mapping = {source_repo_version: dest_repo_version}
+        # TODO: a more structured way to store this state would be nice.
+        content_to_copy = {}
+        repo_mapping = {}
         libsolv_repo_names = {}
+        base_versions = {}
 
         solver = Solver()
 
-        for src in repo_mapping.keys():
-            repo_name = solver.load_source_repo(src)
-            libsolv_repo_names[repo_name] = src
-
-            content = src.content.filter(content_filter)
-            children = find_children_of_content(content, src)
-            content_to_copy[repo_name] = content | children
-
-        for tgt in repo_mapping.values():
-            solver.load_target_repo(tgt)
+        for entry in config:
```

Comment:
There's a little bit of code duplication here now, but I don't see a good way around it that wouldn't end up equally or more complex / messy.

Reply:
If we pull the duplicate code (say, 118..133) out into a method that takes 'entry' and returns a (source_repo_version, dest_repo, dest_repo_version, content_filter) tuple, we can at least keep the duplicate code in one place

Reply:
Line 128 differs from what's up above which makes it a little less straightforwards.

Reply:
If we move 128 to 134 (outside the area of duplicate code) we should be good, yeah?